packages/data-stores/src/help-center/reducer.ts

```diff
@@ -51,9 +51,7 @@ const isMinimized: Reducer< boolean, HelpCenterAction > = ( state = false, actio
 };
 
 const site: Reducer< HelpCenterSite | undefined, HelpCenterAction > = ( state, action ) => {
-	if ( action.type === 'HELP_CENTER_RESET_STORE' ) {
-		return undefined;
-	} else if ( action.type === 'HELP_CENTER_SET_SITE' ) {
```

Comment:
Thanks for weighing in, I appreciate your expert opinion 🙇  I definitely don't want to cause regressions by this change, but also don't want to complicate the code if we can avoid it.

Let's maybe take a step back and let me explain my thinking. I need to lean on your expertise and knowledge of Help Center to untangle this 🙏
Right now it feels like `site` is not managed correctly. The fact that it should and should not be cleared when other fields are OK to be cleared are my "red flags" here. If we further solidify that by making it "official" to sometimes exclude site from clearing the store, it will just create ambiguity and subtle bugs.

I think the root issue is how `site` is set to begin with. Looks like it's here:

https://github.com/Automattic/wp-calypso/blob/trunk/packages/help-center/src/components/help-center.tsx#L105

But what's missing is what happens when it's cleared. That code is not reacting to that state. So once the site is cleared (by submitting the support request, for example), `site` remains `undefined`. Even if the user switches to another site. Only a refresh helps, it seems.
Looks like maybe that line is missing a hook dependency? But the best I can come up with is adding `site` (the one from the Help Center store) there... but the hook itself is modifying that value, so it feels iffy.

We could also maybe handle this in the reducer when clearing the site? And immediately set a "default" value there (current, primary or the one from `window`, cause _of course_ Atomic is not making things easy ;)).

Reply:
What a beautiful explanation. Thank you so much. Yeah syncing Calypso store with the Help Center store is quite iffy. I'll try some things and report back shortly.

Reply:
> Looks like maybe that line is missing a hook dependency? But the best I can come up with is adding site (the one from the Help Center store) there... but the hook itself is modifying that value, so it feels iffy.

I agree but honestly I would go for this. It's a more accurate representation of the intention of syncing the site from Calypso state to the Help Center state.

I tested it [here](https://github.com/Automattic/wp-calypso/pull/85811), and it doesn't create an infinite loop because the equality check passes and the effect doesn't run more than it needs to. Obviously, it's a tradeoff and I leave it to you to decide which option is better.

Reply:
Thanks for the thorough help and testing 🙇‍♂️ And the vote of confidence - if it sounds good to you, then let's go for it. I've updated this PR. I did not notice your PR first, so I used `storedSite` as the name of the site we store in the Help Center store, but let me know if you think `selectedSite` is a better variable name. We have quite a few "sites" in that context. `selectedSite` makes it seem like this is the "normal" selected site from Calypso, but it's a bit more complicated than that. On the other hand, it's the site that _has_ been `select`ed 😅 `selectedStoredSite`? 😆

Reply:
I'm glad you didn't. I like `storedSite` better than `selectedSite`. I meant to say "stored" but the word didn't occur to me.